addons/web/static/src/core/utils/autoresize.js

```diff
@@ -0,0 +1,74 @@
+/** @odoo-module **/
+
+import { useRefListener } from "./hooks";
+import { useEffect } from "@odoo/owl";
+
+/**
+ * This is used on text inputs or textareas to automatically resize it based on its
+ * content each time it is updated. It takes the reference of the element as
+ * parameter and some options. Do note that it may introduce mild performance issues
+ * since it will force a reflow of the layout each time the element is updated.
+ *
```

Comment:
not sure, but if i understand it properly, with textareas, it only works if it is the only child from its parent element. If that is true, maybe add a comment to say that